indigo_app/static/javascript/indigo/views/annotations.js

```diff
@@ -535,76 +512,58 @@
       target = Indigo.dom.rangeToTarget(this.pendingRange, this.contentContainer);
       if (!target) return;
 
+      let focusViewInput = false;
+      const layoutCompleteHandler = () => {
+        focusViewInput = true;
+        this.gutter.removeEventListener('layoutComplete', layoutCompleteHandler);
+      }
+
+      this.gutter.addEventListener("layoutComplete", layoutCompleteHandler)
+
+      // createThread triggers add, the makeView which appends la-gutter-item
       thread = this.threads.createThread({selectors: target.selectors, anchor_id: target.anchor_id, closed: false});
 
+      // this runs straight after makeView, and this.gutter.runLayout is a 200ms delay from debounce
       this.threadViews.forEach(v => {
         if (v.model === thread) {
           v.display();
 
-          this.gutter.activeItem = v;
-          // once it's rendered in the gutter, focus it
-          this.gutter.$nextTick(() => {
-            v.focus();
-            v.$el
-             .find('textarea')
-             .first()
-             .focus();
-          });
-        } else {
-          v.blur();
+          v.el.active = true;
+          //Wait for layout to be finished, so view is in correct position, then focus text area
+          return new Promise((resolve) => {
+            const interval = setInterval(() => {
+              if(focusViewInput) {
+                resolve();
+                clearInterval(interval);
+              }
+            },100);
+          }).then(() => {
+            v.el.querySelector('textarea:first-child').focus();
+          })
         }
       });
 
     },
 
     removeNewButton: function() {
-      const ix = this.gutter.items.indexOf(this.newButtonItem);
-    if (ix > -1) {
-      this.gutter.items.splice(ix, 1);
-    }
-    this.newButtonTimeout = null;
+      this.newButtonItem.remove();
+      this.newButtonTimeout = null;
     },
 
     nextAnnotation: function(e) {
       e.preventDefault();
       e.stopPropagation();
-
-      this.scrollSelected(true);
+      this.gutter.activateNextItem().then((activeItem) => {
+        activeItem.scrollIntoView({ behavior: "smooth" });
```

Comment:
use an alternative method instead of scrollIntoView. `scrollIntoView` does not have `offsetTop` capabilities. I have tried 
these techniques
https://matthiasott.com/notes/scrolling-elements-into-view
no luck yet.
I have to pop right now, this is on my todo list to change

Reply:
Hmm. Double check it's even necessary -- the new annotation should already be visible because we've just highlighted the text that forms the anchor for it.

Reply:
> Hmm. Double check it's even necessary -- the new annotation should already be visible because we've just highlighted the text that forms the anchor for it.

Ignore this -- wrong place :)

What does scrollSelected do? Can we just do that?

Reply:
used logic for scrollSelected
https://www.loom.com/share/7a9d56b812f34c3482841ccabef17c24

Reply:
Smooth like butter 🧈